lib/danom/monad.rb

```diff
@@ -51,15 +51,15 @@ def ~@
     end
 
 
-    # Calls #to_s on the underlying value.
+    # Proxy all Object methods to underlying value
     #
-    # @note This happens because method_missing does not override #to_s
     # @private
-    def to_s
-      and_then { |v| v.to_s }
+    Object.new.methods.each do |object_method|
+      define_method object_method do |*args|
+        and_then { |v| v.public_send(object_method, *args) }
```

Comment:
Danom::Monad has the variable name 'v'

[Read more about it here.](https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Variable-Name.md)